fetch.js

```diff
@@ -274,7 +274,17 @@ function Body() {
 
     this.arrayBuffer = function() {
       if (this._bodyArrayBuffer) {
-        return consumed(this) || Promise.resolve(this._bodyArrayBuffer)
+        consumed(this)
```

Comment:
Consumed returns either `undefined` or a rejected Promise.

The previous code returned the rejected Promise.
This changes the original behaviour.
Was that intended?